examples/t_cose_basic_example_ossl.c

```diff
@@ -528,17 +538,195 @@ int two_step_sign_example(void)
+ *
+ * The signing operation of t_cose requires the caller to provide a
+ * buffers large enough to hold the result. If the provided buffer is
+ * too small, the operation will fail.
+ *
+ * When EDDSA is used, an additional auxiliary buffer is needed for
+ * both signing and verification.
+ *
+ * While memory-constrained applications may want to use stack or
+ * statically allocated buffers of a fixed size, others prefer the
+ * flexibility of dynamically allocate buffers of the right size on
+ * demand.
+ *
+ * This example shows how to call t_cose to determine the size of the
+ * output and auxiliary buffers, before dynamically allocating them
+ * using malloc and free.
+ *
+ */
+int32_t dynamic_buffer_example(void)
+{
+    struct t_cose_sign1_sign_ctx   sign_ctx;
+    enum t_cose_err_t              return_value;
+    struct q_useful_buf            signed_cose_buffer;
+    struct q_useful_buf_c          signed_cose;
+    struct q_useful_buf            auxiliary_buffer;
+    struct q_useful_buf_c          constructed_payload;
+    struct q_useful_buf_c          returned_payload;
+    struct t_cose_key              key_pair;
+    struct t_cose_sign1_verify_ctx verify_ctx;
+
+    /* ------   Prepare the payload   ------ */
+    constructed_payload = Q_USEFUL_BUF_FROM_SZ_LITERAL("payload");
+    printf("Encoded payload size = %ld\n", constructed_payload.len);
+
+
+    /* ------   Make an EDDSA key pair    ------ */
+    return_value = make_ossl_key_pair(T_COSE_ALGORITHM_EDDSA, &key_pair);
+    printf("Made EC key with curve ed25519: %d (%s)\n", return_value, return_value ? "fail" : "success");
+    if(return_value) {
+        goto Done;
+    }
+
+    /* ------   Initialize for signing    ------ */
+    t_cose_sign1_sign_init(&sign_ctx, 0, T_COSE_ALGORITHM_EDDSA);
+    t_cose_sign1_set_signing_key(&sign_ctx, key_pair,  NULL_Q_USEFUL_BUF_C);
+
+    /* ------   Compute the size of the output and auxiliary buffers   ------
+     *
+     * A large but NULL output buffer is given to the signing operation.
+     * The size of result, signed_cose, will reflect how big of a buffer
+     * needs to be provided for the real operation.
+     *
+     * Similarly, the necessary auxiliary buffer size is saved in the
+     * signing context and available by calling t_cose_sign1_sign_auxiliary_buffer_size.
+     *
+     * Both sizes are used later on to allocate the proper buffers.
+     */
+    return_value = t_cose_sign1_sign(&sign_ctx,
+                                      constructed_payload,
+                                      (struct q_useful_buf){ NULL, SIZE_MAX },
+                                     &signed_cose);
+    printf("Computed signing size %d (%s)\n", return_value, return_value ? "fail" : "success");
+    if(return_value) {
+        goto Done;
+    }
+    printf("Output buffer size = %d bytes\n", signed_cose.len);
+    printf("Auxiliary buffer size = %d bytes\n", t_cose_sign1_sign_auxiliary_buffer_size(&sign_ctx));
```

Comment:
%zu to print size_t  Here and a few other places.

Reply:
Done.